control-plane/config-entries/controllersv2/gateway_class_controller.go

```diff
@@ -27,7 +27,7 @@ type GatewayClassController struct {
 // +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=gatewayclass/status,verbs=get;update;patch
 
 func (r *GatewayClassController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	return r.Controller.ReconcileEntry(ctx, r, req, &meshv2beta1.GatewayClass{})
+	return ctrl.Result{}, nil
```

Comment:
GatewayClass was not registered with the consul API and was throwing errors.

Reply:
> [!NOTE]
> This change is being handle separately in #3286